### PR TITLE
Smaller SRF Head: hidden 128/96 vs baseline 192

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

PR #2170 showed that increasing SRF hidden dim (192→256/384) overfits: p_tan +4-5%. If adding SRF capacity hurts OOD transfer, maybe h=192 isn't the sweet spot — maybe smaller is better.

**Rationale:** Every experiment that adds parameters or capacity has regressed p_tan. The baseline sits at a precise optimum. Reducing SRF capacity provides implicit regularization — the head has LESS room to memorize training-distribution surface pressure patterns and MUST learn more generalizable corrections.

This is the logical inverse of frieren's experiment. If h=256 is worse and h=192 is current, we should test h=128 and h=96 to find whether the optimum is actually below 192.

## Instructions

Run **2 seeds (42, 73)** for **each of 2 configs**, using `--wandb_group tanjiro-smaller-srf`:

**Config A — h=128:**
```bash
cd cfd_tandemfoil && python train.py --agent tanjiro \
  --wandb_name "tanjiro/srf-h128" --wandb_group tanjiro-smaller-srf \
  --surface_refine_hidden 128 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias \
  --seed 42
```
(repeat with `--seed 73`)

**Config B — h=96:**
Same command with `--surface_refine_hidden 96` and `--wandb_name "tanjiro/srf-h96"`. (Repeat with `--seed 73`.)

Run all 4 in parallel. Report p_in, p_oodc, p_tan, p_re per config (2-seed avg).

## Baseline

| Metric | Baseline (h=192, 2-seed avg) |
|--------|------------------------------|
| p_in | 13.05 |
| p_oodc | 7.70 |
| **p_tan** | **28.60** |
| p_re | 6.55 |

Baseline W&B runs: d7l91p0x (seed 42), j9btfx09 (seed 73)

**Reproduce baseline:**
```bash
cd cfd_tandemfoil && python train.py --agent <name> --wandb_name "<name>/baseline-gsb-pcgrad" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT tanjiro:

## Results

All 4 experiments completed (180-181 min each, timeout-limited at epoch 149-151).

### Per-run results

| Run | Epoch | val_loss | p_in | p_tan | p_oodc | p_re | W&B |
|-----|-------|----------|------|-------|--------|------|-----|
| h=128, s42 | 150 | 0.3875 | 13.0 | 30.1 | 7.7 | 6.5 | 281pher3 |
| h=128, s73 | 151 | 0.3883 | 13.3 | 29.8 | 7.7 | 6.6 | w6apleaz |
| h=96, s42 | 150 | 0.3879 | 13.2 | 30.2 | 7.8 | 6.5 | uoaonrpq |
| h=96, s73 | 149 | 0.3858 | 13.0 | 29.2 | 7.9 | 6.5 | u9a830l5 |

### 2-seed averages vs baseline (h=192)

| Config | p_in | p_tan | p_oodc | p_re | Δ p_tan |
|--------|------|-------|--------|------|---------|
| **Baseline (h=192)** | **13.05** | **28.60** | **7.70** | **6.55** | — |
| h=128 | 13.15 | 29.95 | 7.70 | 6.55 | +4.7% worse |
| h=96 | 13.10 | 29.70 | 7.85 | 6.50 | +3.8% worse |

### Analysis

**Hypothesis rejected.** Both smaller SRF head sizes are worse than baseline h=192 on p_tan.

**h=128** (p_tan=29.95): +4.7% worse. The regularization from reduced capacity didn't help — the head needs the full 192 dimensions to learn effective surface corrections.

**h=96** (p_tan=29.70): +3.8% worse, but interestingly slightly better than h=128. The h=96 s73 run (29.2) is the best individual run, close to baseline s73 (28.3). High seed variance suggests h=96 is in a noise-dominated regime — sometimes it gets lucky but isn't reliably better.

**Key observations:**
1. **p_in and p_re are nearly unchanged** across all configs (~13.1, ~6.5), suggesting the SRF head doesn't strongly affect in-distribution or Reynolds OOD — its primary impact is on tandem transfer.
2. **p_oodc is also stable** (7.7-7.9), confirming SRF head size mainly matters for tandem.
3. **The SRF head capacity sweet spot is at h=192.** Smaller (96, 128) and larger (256, 384 from PR #2170) both regress p_tan. This is classic bias-variance tradeoff: too small → underfits surface corrections, too large → overfits.
4. **Combined with PR #2170 (h=256/384 worse) and these results (h=96/128 worse), h=192 is confirmed as optimal.** No further SRF head size sweeps are needed.

### Suggested follow-ups

- SRF head size is confirmed at h=192. Further gains should come from how the head is trained (loss formulation, gradient flow) rather than its capacity.
- The stable p_in/p_re across all SRF sizes suggests those metrics are bottlenecked elsewhere (backbone capacity, data representation).
- p_tan improvement likely needs fundamentally different approaches to tandem transfer — architecture changes that affect how the model represents wake interactions, not just how it refines surface predictions.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2178 because ADVISOR: Closing — both smaller SRF sizes regress p_tan.

**h=128 avg:** p_tan=29.90 (+4.5%), p_oodc=7.71, p_in=13.12, p_re=6.57
**h=96 avg:** p_tan=29.71 (+3.9%), p_oodc=7.82, p_in=13.10, p_re=6.51

Combined with PR #2170 (h=256/384 also worse), h=192 is **definitively confirmed optimal** — classic bias-variance sweet spot. Smaller underfits surface corrections, larger overfits.

Your analysis is excellent. The stable p_in/p_re across all SRF sizes is an important finding — those metrics are bottlenecked by the backbone, not the SRF head. Future gains need to come from how information reaches the SRF head (input features, backbone representations), not the head's capacity.

SRF head capacity is now fully characterized: 96 < 128 < 256 < 384 < **192 (optimal)**. No further sweeps needed..
